src/components/CallView/CallView.vue

```diff
@@ -1,75 +1,310 @@
+		</div>
+		<div id="screens">
+			<Screen v-if="localMediaModel.attributes.localScreen"
+				:local-media-model="localMediaModel"
+				:shared-data="localSharedData" />
+			<Screen v-for="callParticipantModel in callParticipantModelsWithScreen"
+				:key="'screen-' + callParticipantModel.attributes.peerId"
+				:call-participant-model="callParticipantModel"
+				:shared-data="sharedDatas[callParticipantModel.attributes.peerId]" />
 		</div>
 	</div>
 </template>
 
 <script>
-import Avatar from '@nextcloud/vue/dist/Components/Avatar'
-import MediaControls from './MediaControls'
+import LocalVideo from './LocalVideo'
+import Screen from './Screen'
+import Video from './Video'
+import LocalMediaModel from '../../utils/webrtc/models/LocalMediaModel'
+import LocalCallParticipantModel from '../../utils/webrtc/models/LocalCallParticipantModel'
+import CallParticipantCollection from '../../utils/webrtc/models/CallParticipantCollection'
 
 export default {
+
 	name: 'CallView',
 
 	components: {
-		Avatar,
-		MediaControls,
+		LocalVideo,
+		Screen,
+		Video,
+	},
+
+	props: {
+		token: {
+			type: String,
+			required: true,
+		},
+		signaling: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			speakers: [],
+			speakingUnwatchers: {},
+			screens: [],
+			screenUnwatchers: {},
+			// callParticipantModelsWithScreen: [],
+			localSharedData: {
+				screenVisible: true,
+			},
+			sharedDatas: {},
+
+			localMediaModel: {},
+			localCallParticipantModel: {},
+			callParticipantCollection: {
+				callParticipantModels: [],
+			},
+		}
 	},
 
 	computed: {
-		actorType() {
-			return this.$store.getters.getActorType()
+
+		callViewClass() {
+			const callViewClass = {
+				'incall': this.remoteParticipantsCount > 0,
+				'screensharing': this.screenSharingActive,
+			}
+			callViewClass['participants-' + (this.remoteParticipantsCount + 1)] = true
+
+			return callViewClass
 		},
-		actorId() {
-			return this.$store.getters.getActorId()
+
+		reversedCallParticipantModels() {
+			return this.callParticipantCollection.callParticipantModels.slice().reverse()
 		},
-		displayName() {
-			return this.$store.getters.getDisplayName()
+
+		remoteParticipantsCount() {
+			return this.callParticipantCollection.callParticipantModels.length
+		},
+
+		callParticipantModelsWithScreen() {
+			return this.callParticipantCollection.callParticipantModels.filter(callParticipantModel => callParticipantModel.attributes.screen)
+		},
+
+		screenSharingActive() {
+			return this.screens.length > 0
 		},
 
-		firstLetterOfGuestName() {
-			const customName = this.displayName !== t('spreed', 'Guest') ? this.displayName : '?'
-			return customName.charAt(0)
+		darkBackground() {
+			return this.remoteParticipantsCount > 0 || this.screenSharingActive
 		},
+
 	},
-}
-</script>
 
-<style lang="scss" scoped>
-.stage {
-	background: #000000;
-	width: 100%;
-	height: 100%;
+	watch: {
+
+		'darkBackground': function(darkBackground) {
```

Comment:
you do not need the quotes :)
```suggestion
		darkBackground: function(darkBackground) {
```